src/ScheduleFunctions.cpp

```diff
@@ -45,9 +45,11 @@ struct Container {
     int dim_idx;
     string name;
     Expr value;
+    // Only meaningful if type==For.
+    ForType for_type;
```

Comment:
This looks like it could be const.

Reply:
I think const members might be a bad idea - you can't move from them, which means the whole class doesn't get a default move constructor.

Reply:
Making this (or other members) const requires changing large amount of code. In particular there are several std::swap() calls with Containers.